include/systems/inter_mesh_projection.h

```diff
@@ -87,7 +87,7 @@ class GradientMeshFunction : public FunctionBase<Gradient>
 
   virtual std::unique_ptr<FunctionBase<Gradient>> clone () const
   {
-    return libmesh_make_unique<GradientMeshFunction>(mesh_function.get());
+    return libmesh_make_unique<GradientMeshFunction>(*(mesh_function.get()));
```

Comment:
Just ```*mesh_function``` will do here; ```unique_ptr``` defines ```operator*``` so you don't need to manually ```get()``` the underlying pointer.